src/sentry/incidents/models/incident.py

```diff
@@ -97,28 +108,33 @@ def get_active_incident(self, alert_rule, project):
 
     @classmethod
     def clear_active_incident_cache(cls, instance, **kwargs):
+        # instance is an Incident
         for project in instance.projects.all():
-            cache.delete(cls._build_active_incident_cache_key(instance.alert_rule_id, project.id))
-            assert (
-                cache.get(cls._build_active_incident_cache_key(instance.alert_rule_id, project.id))
-                is None
-            )
+            subscription = instance.subscription
+            if subscription:
+                key = cls._build_active_incident_cache_key(
+                    instance.alert_rule_id, project.id, subscription.id
+                )
+            else:
+                key = cls._build_active_incident_cache_key(instance.alert_rule_id, project.id, None)
+            cache.delete(key)
+            assert cache.get(key) is None
```

Comment:
You could get burned here by concurrent requests.

Reply:
I'm not totally sure why this get is here, it looks like it's meant to validate that the cache clear worked, but not sure why it's in code vs a test

Reply:
I think we can leave it. It's not causing a problem today I think?